x/stakeibc/keeper/unbonding.go

```diff
@@ -247,147 +273,43 @@ func (k Keeper) GetUnbondingICAMessages(
 		}
 		remainingUnbondAmount = remainingUnbondAmount.Sub(unbondAmount)
 
-		// Build the validator splits for the callback
-		unbondings = append(unbondings, &types.SplitDelegation{
-			Validator: validatorCapacity.ValidatorAddress,
-			Amount:    unbondAmount,
-		})
-	}
-
-	// If the number of messages exceeds the batch size, shrink it down the the batch size
-	// by re-distributing the exceess
-	if len(unbondings) > batchSize {
-		unbondings, err = k.ConsolidateUnbondingMessages(totalUnbondAmount, unbondings, prioritizedUnbondCapacity, batchSize)
-		if err != nil {
-			return msgs, unbondings, errorsmod.Wrapf(err, "unable to consolidate unbonding messages")
+		// Calculate the associated stToken amount
+		// There may be some precision error so the last validator should
+		// get all the remaining stTokens
+		var stAmount sdkmath.Int
+		if remainingUnbondAmount.IsZero() {
+			stAmount = remainingStAmount
```

Comment:
This guarantees that remainingStAmount will hit zero at the same time that remainingUnbondAmount reaches zero so there is no need for a break on the remainingStAmount being zero or even a sanity check for it.